qiskit/transpiler/passes/routing/algorithms/bip_model.py

```diff
@@ -0,0 +1,408 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Integer programming model for quantum circuit compilation."""
+
+import logging
+
+from docplex.mp.model import Model
```

Comment:
You need to check whether docplex is installed or not because docplex is not in requirements.txt but in requirements-dev.txt.

Reply:
Addressed at 761745d